leakcanary-android-core/src/main/java/leakcanary/internal/activity/LeakViews.kt

```diff
@@ -63,7 +64,11 @@ internal fun View.shareToStackOverflow(content: String) {
   Toast.makeText(context, R.string.leak_canary_leak_copied, Toast.LENGTH_LONG)
       .show()
   val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(STACKOVERFLOW_QUESTION_URL))
-  activity.startActivity(browserIntent)
+  try {
+    activity.startActivity(browserIntent)
```

Comment:
AFAIK a correct way to handle this is by resolving activity:
https://developer.android.com/guide/components/intents-filters#ExampleSend

I'm curious, what kind of device you're testing on that doesn't have a browser? It's not an impossible scenario, but just strange not to have even built-in browser :)

Reply:
1. I think we need to tell the user what's going wrong by a `Toast`, we can just reuse the exception message by using `try..catch..`. On the other hand, if we use `android.content.Intent#resolveActivity`, we have to custom a message.
2. Android-based infotainment system that is built in to vehicles

Reply:
Custom message will do the job
```
if (sendIntent.resolveActivity(packageManager) != null) {
    startActivity(sendIntent)
} else {
  Toast.makeText(context, R.string.leak_canary_leak_missing_browser_error, Toast.LENGTH_LONG)
      .show()
```

Reply:
> Android-based infotainment system that is built in to vehicles
Wow, so you're using LeakCanary to catch leaks in the cars? We should start advertising this library as a good car plumbing solution! 😄

Reply:
There are changes related to package visibility in Android 11: https://developer.android.com/preview/privacy/package-visibility
so `resolveActivity` won't work anymore in this case. I think `try-catch` is the only solution that will work on all API levels.

@Armaxis you can test it on any device. You don't have to uninstall the browser, disabling it is enough.

Reply:
Wow, good catch! Yeah, looks like exception handling [is a correct way from now on](https://developer.android.com/preview/privacy/package-visibility-use-cases#open-urls-browser-or-other-app)
Google is breaking APIs again *sigh*